packages/core/src/utils/operation.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  GraphQLRequest,
+  Operation,
+  OperationContext,
+  OperationType,
+} from '../types';
+import { Warning, deprecationWarning } from './deprecation';
+
+// TODO: Remove when the deprecated `operationName` property is removed
+const DEPRECATED: Record<string, Warning> = {
+  operationName: {
+    key: 'Operation.operationName',
+    message:
+      'The "Operation.operationName" property has been deprecated and will be removed in a future release of urql. Use "Operation.kind" instead.',
+  },
+};
+
+export const makeOperation = (
+  kind: OperationType,
+  request: Pick<GraphQLRequest, 'key' | 'query' | 'variables'>,
+  context: OperationContext
+): Operation => ({
+  key: request.key,
+  query: request.query,
+  variables: request.variables,
+  kind,
+  context,
+
+  get operationName(): OperationType {
+    deprecationWarning(DEPRECATED.operationName);
```

Comment:
Last thing, I swear 😅 Let's wrap this in `if (process.env.NODE_ENV !== 'production')`

We can then check in the output `min.js` bundles whether all warnings successfully compile away in our output

Reply:
Should I wrap it here, or inside the `deprecationWarning` util function itself? The question is essentially "one place, or many places"?

Reply:
Nice! So - `(name === 'warn' || name === 'deprecationWarning')` then?